drake/core/Core.h

```diff
@@ -6,44 +6,110 @@
 #include <Eigen/Dense>
 
 namespace Drake {
-
-#include "Path.h"
-
-// useful refs on generic programming: http://www.generic-programming.org/languages/cpp/techniques.php
-//
-// Concept: Vector<ScalarType>.
-// Requirements:
-//  - Defines a static constant (or enum) RowsAtCompileTime (see Pendulum.h for an example).  Can be Eigen::DYNAMIC
-//  - implements size_t size(), which equals RowsAtCompileTime except when DYNAMIC
-//  - implements the copy constructor and the assignment operator= from const Eigen::MatrixBase<Derived>& (e.g. using "copy and swap")
-//  - implements the typecast method to convert back to an Eigen Vector (e.g.  template <int Rows> operator Eigen::Matrix<ScalarType,Rows,1>())
-//  - implements std::ostream& operator<<(std::ostream& os, const Vector& x)
-// (The methods/defs below make it possible (and hopefully easy) to simply use Eigen::Matrix types to model this concept)
-
-
-  // In order to use Eigen types to model the Drake::Vector concept, they must accept only a single template parameter (ScalarType)
-  // The following structures provide a means for doing it. (e.g. via Drake::VectorBuilder<Rows>::VecType )
-
+  #include "Path.h"
```

Comment:
Should probably move outside of namespace. See https://stackoverflow.com/questions/2868955/shielding-include-within-namespace-block/2868971#2868971.